frontend/src/metabase/query_builder/components/DataSelector/DataSelector.jsx

```diff
@@ -662,6 +665,12 @@ export class UnconnectedDataSelector extends Component {
   async loadStepData(stepName) {
     const loadersForSteps = {
       // NOTE: make sure to return the action's resulting promise
+      [DATA_BUCKET_STEP]: () => {
+        return Promise.all([
+          this.props.fetchDatabases(this.props.databaseQuery),
+          this.props.fetchDatabases({ saved: true }),
+        ]);
+      },
```

Comment:
After refactoring, [this test](https://github.com/metabase/metabase/blob/75b6d92/e2e/test/scenarios/question/notebook.cy.spec.js#L504-L517) started to fail, and these changes make it pass again.